### ISSUE TITLE
[Question]: Is grouping multiple inputs with one label consider accessible?

### ISSUE THREAD

**Issue by grahamharper** [IBMa/equal-access#1858]
### Project

able site

### Description

The Content Design guidance on the Able site illustrates an example where a single label is used for multiple inputs, and the actual label for each input is specified as a hidden aria-label, which is only available to screenreader users, not sighted users.

![image](https://github.com/IBMa/equal-access/assets/2853295/7c61ecf5-d087-4b52-a4e6-a274fd32519c)

whereas directly above this image the guidance states to "Provide clear and concise labels for every input element"

![image](https://github.com/IBMa/equal-access/assets/2853295/8b1bf019-ec00-40dc-9fdb-83df62420b70)

without a visible label for each input, how would a sighted user know what each input in the phone number group is for?

**Comment by mbgower** [comment]
> without a visible label for each input, how would a sighted user know what each input in the phone number group is for?

Can we agree that mutli-part inputs with one label exist i some designs? Obvious examples are "Address" and "Phone". 
I'm not defending that design decision, but the guidance IS saying that where someone elects to make one label visually apply to multiple inputs, the implementation team needs to ensure that that label is applied programmatically to both of them.

If you can think of a succinct way of saying that which makes it easier for you to understand, we're happy to incorporate.

**Comment by grahamharper** [comment]
> Can we agree that mutli-part inputs which only have one label exist in some designs?

Yes. I support an application that currently has many of them (which I've now inherited) that use the technique describe in that diagram but which had a11y defects opened against them claiming they are not accessible to sighted keyboard users and touch users (which I would agree with).

But I was curious to know why I've seen this pattern described here and in [Technique H65](https://www.w3.org/WAI/WCAG21/Techniques/html/H65) which on first look might read as an endorsement of the pattern as fully accessible to all, so trying to understand what I might be missing.

So I'm thinking the piece I'm missing is that marking up this pattern as described is a way to make this accessible _to- screenreaders/assistive technology_ but does not mean it is accessible to _all_ users i.e. sighted, keyboard-only/touch users.

So when you say

> the answer is that if they are implemented following the guidance you cited in your first illustration, they can be made accessible.

would you qualify that by adding "…they can be made accessible, _to assistive technology_"?

If I'm right then more qualified guidance for designers might something along the lines of.

> Provide clear and concise labels for every input element.
> Annotate any labels that apply to more than one input field.
> Avoid hiding input labels so the purpose of each input is clear to all users. If input labels are hidden, annotate hidden aria labels where each input lacks a unique, visible label so they are available to assistive technology users. Note they may not be available to other users.

**Comment by mbgower** [comment]
>  on first look might read as an endorsement of the pattern as fully accessible to all, so trying to understand what I might be missing.

A sufficient technique is one way to make something that is failing a requirement sufficiently meet it in one context. That's why techniques will often be categorized under Scenarios or Situations. They should most definitely not be interpreted as a best practice. There is a bunch of context offered in techniques about this, generally.

It's also worth noting that there is no accessibility requirement that explicitly covers users without disabilities (although many could improve the experience of everyone). Something that is a problem for _everyone_ is generally regarded as a design problem, not an accessibility problem. Obviously there are nuances to this (i.e., something can affect everyone, but it can have a disproportionate affect on users with disabilities).

> does not mean it is accessible to all users i.e. sighted, keyboard-only/touch users.

So, in regard to someone labelling two inputs like this:
Address:
[one input]
[second input]

With the proviso that it is unlikely to win a design award and has problems, IF you encounter this as a developer (i.e., there is nothing I can do to change this design I have to implement), from an accessibility perspective:
1. there is no obvious issue for someone whose only challenge is they rely on a keyboard. The user can tab through the inputs fine
2. there is no non-magnified visibility issue (the position of the label over the first input provides the same information to any user with typical vision).
3. there is a potential issue for low vision users using magnification (where they magnify to the point where the label is not in view when they are at the second input). One way of improving this to some extent is to use title attributes to expose the label for someone who is using a mouse*
4. There is an obvious issue for someone who cannot see. They cannot tell that the inputs are both positioned underneath a single label, which by inference applies to them both. This is addressed by making each input programmatically associated with the one label.

* In regard to H65, use of title as a stop gap in a design is controversial (to the point where [H65](https://www.w3.org/WAI/WCAG22/Techniques/html/H65) not even listed under [Labels or Instructions](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions.html) as sufficient in WCAG 2.2). If you do a search for H65, you'll see lots of issues, both open and closed about it. I'd suggest that is a whole other topic, and we don't want to go off on a tangent.

**Comment by grahamharper** [comment]
@mbgower Thank you for that really helpful response. It has added some much needed context.

**Comment by shunguoy** [comment]
Triage: this is a toolkit issue, and will need more discussion.

**Comment by mbgower** [comment]
@grahamharper I think a primary 'glitch' here is that the a11y guidance _started_ as dev remediation, and now that we've picked it up and put it in design, it _appears_ that we're advocating using only one labe on multiple inputs. This is partially a result of some consolidation/simplification of our design guidance that occurred a couple years back. I've flagged your issue, and we're aware of the challenge. Thanks!
